### ISSUE TITLE
SQLite dependency resolution fails

### ISSUE THREAD

**Issue by albemala** [aws-amplify/amplify-ios#1873]
### Describe the bug

I'm having a problem with the updated SQLite dependency.

My project depends on
pod 'Amplify', '1.19.1'
pod 'AWSAppSync', '3.6.1'

If I try to upgrade Amplify to a version higher than 1.22.2, I get the following error

[!] CocoaPods could not find compatible versions for pod "SQLite.swift":
  In snapshot (Podfile.lock):
    SQLite.swift (= 0.12.2, ~> 0.12.2)

  In Podfile:
    AWSAppSync (= 3.6.1) was resolved to 3.6.1, which depends on
      SQLite.swift (~> 0.12.2)

    AmplifyPlugins/AWSDataStorePlugin (= 1.24.0) was resolved to 1.24.0, which depends on
      SQLite.swift (= 0.13.2)

Basically,
AWSAppSync depends on SQLite.swift = 0.12.2
AmplifyPlugins depends on SQLite.swift = 0.13.2
So Pods dependency resolution fails.

Should I open a ticket for AWSAppSync to upgrade SQLite.swift? Is there another solution?

Thank you for your help!

### Steps To Reproduce

```swift
...
```

### Expected behavior

...

### Amplify Framework Version

1.22.2

### Amplify Categories

API, Auth, DataStore, Storage

### Dependency manager

Cocoapods

### Swift version

5.4

### CLI version

6.4.0

### Xcode version

13.3

### Is this a regression?

Yes

### Device

iPhone, iPad

### iOS Version

14, 15

**Comment by atierian** [comment]
Thanks for opening this issue @albemala.
Adding some findings:

### Repro Steps
1. Add these pods to Podfile.
```ruby
pod 'Amplify', '1.22.3'
pod 'AmplifyPlugins/AWSDataStorePlugin', '1.22.3'
pod 'AWSAppSync', '3.6.1'
```
2. run `pod install`

#### Results in:
```
[!] CocoaPods could not find compatible versions for pod "SQLite.swift":
  In snapshot (Podfile.lock):
    SQLite.swift (= 0.12.2, ~> 0.12.2)

  In Podfile:
    AmplifyPlugins/AWSDataStorePlugin (= 1.22.3) was resolved to 1.22.3, which depends on
      SQLite.swift (= 0.13.2)

Specs satisfying the `SQLite.swift (= 0.12.2, ~> 0.12.2), SQLite.swift (= 0.13.2)` dependency were found, but they required a higher minimum deployment target.
```
(The same error occurs when installing pods without an explicit version)

### Additional Information

The issue exists with Cocoapods because of the discrepancy in the respective `Podfile.lock` 
[[AWS AppSync SDK for iOS](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/blob/main/Podfile.lock#L19)] 
[[AmplifyPlugins/AWSDataStorePlugin](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/DataStore/Podfile.lock#L53)]

This isn't an issue when using SPM. The SQLite.swift dependency resolves to `0.13.2` because of the `from:` rule in `AWS AppSync SDK for IOS` [package manifest](https://github.com/awslabs/aws-mobile-appsync-sdk-ios/blob/main/Package.swift#L26). 

### Next Steps
- See if updating the SQLite.swift dependency in `AWS AppSync SDK for iOS` introduces any breaking changes.

**Comment by albemala** [comment]
Thank you @atierian for looking into this

**Comment by undefobj** [comment]
@albemala it looks like you are trying to use the [AWS AppSync SDK](https://cocoapods.org/pods/AWSAppSync) along with the Amplify categories of API or DataStore in your application. This isn't a supported configuration, we recommend customers use API or DataStore category if they are able and not the AppSync SDK unless they have a specific requirement to use that. Could you see if that fits your needs better?

**Comment by alharris-at** [comment]
@albemala +1 to what Richard has said above, if you have additional questions or concerns that either Amplify API https://docs.amplify.aws/lib/graphqlapi/getting-started/q/platform/ios/ or Datastore https://docs.amplify.aws/lib/datastore/getting-started/q/platform/ios/ doesn't address, feel free to reopen this issue and we can continue discussing.
